jax/lax/lax_control_flow.py

```diff
@@ -339,10 +328,47 @@ def make_computation(name, jaxpr, op_shape):
 
   return c.Conditional(pred, true_op, true_c, false_op, false_c)
 
+def _cond_batching_rule(args, dims, true_jaxpr, false_jaxpr, true_nconsts,
+                        false_nconsts):
+  # TODO: maybe avoid moving arg axes to front if we're promoting to select?
+  args = [batching.moveaxis(x, d, 0) if d is not batching.not_mapped and d != 0
+          else x for x, d in zip(args, dims)]
+  true_nops = len(true_jaxpr.in_avals) - true_nconsts
+  (pred,), true_consts, true_ops, false_consts, false_ops = split_list(
+      args, [1, true_nconsts, true_nops, false_nconsts])
+  size, = {x.shape[d] for x, d in zip(args, dims) if d is not batching.not_mapped}
+  orig_bat = [d is not batching.not_mapped for d in dims]
+  pred_bat, t_bat, tconst_bat, f_bat, fconst_bat = split_list(
```

Comment:
Looks like we missed writing `(pred_bat,)` here, rather than `pred_bat`, since it'll come out of `split_list` as a single-element list (and thus always be truthy, regardless of whether it's `[True]` or `[False]`).

Reply:
Added a test for this.